### PR TITLE
Client library update for new text recognition

### PR DESCRIPTION
**Pull request by mike-int**
Update the client library for new text recognition API is released,

Notes:
=======
This PR just update the client library, after the PR is merged we can use it to create nuget package.

### CODE REVIEW

File: ClientLibrary/Contract/TextRecognitionLine.cs [hunk 1]
```diff
@@ -48,13 +48,21 @@ public class HandwritingTextLine
         /// </value>
         public int[] BoundingBox { get; set; }
```
Comment (cthrash) [inline ClientLibrary/Contract/TextRecognitionLine.cs]: The `BoundingBox` and `Polygon` represent the same information.  This is confusing to the consumer.  Consider the following instead:
```
    public class TextRecognitionWord
    {
        /// <summary>
        /// Gets or sets the text.
        /// </summary>
        /// <value>
        /// The text.
        /// </value>
        public string Text { get; }

        /// <summary>
        /// Gets the polygon
        /// </summary>
        /// <value>
        /// The polygon
        /// </value>
        [JsonIgnore]
        public Polygon Polygon {get;}

        [JsonConstructor]
        public TextRecognitionWord(string text, int[] boundingBox)
        {
            Text = text;
            Polygon = Polygon.FromArray(boundingBox);
        }
    }

```

File: ClientLibrary/Contract/TextRecognitionMode.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+namespace Microsoft.ProjectOxford.Vision.Contract
+{
+    /// <summary>
+    /// The enum of TextRecognitionMode
+    /// </summary>
+    //[JsonConverter(typeof(StringEnumConverter))]
```
Comment (cthrash) [inline ClientLibrary/Contract/TextRecognitionMode.cs]: remove line

File: ClientLibrary/VisionServiceClient.cs [hunk 3]
```diff
@@ -772,7 +774,7 @@ private void HandleException(Exception exception)
 
                             ClientError errorCollection = JsonConvert.DeserializeObject<ClientError>(errorObjectString);
 
-                            // HandwritingOcr error message use the latest format, so add the logic to handle this issue.
+                            // OneOCR error message use the latest format, so add the logic to handle this issue.
```
[the remark this answers is not in the document]
Comment (mike-int) [reply ClientLibrary/VisionServiceClient.cs]: Ok, will update it.

File: ClientLibrary/Contract/TextRecognitionMode.cs (hunk 2 above)
Comment (mike-int) [reply ClientLibrary/Contract/TextRecognitionMode.cs]: Ok

File: ClientLibrary/Contract/TextRecognitionLine.cs (hunk 1 above)
Comment (mike-int) [reply ClientLibrary/Contract/TextRecognitionLine.cs]: Old MSOcr feature use similar format, so recommend to keep it.

**Pull request merged**